Control more than one Philips Hue light

grunskis opened the pull request:
1) During on-boarding all the available lights from an authenticated bridge would be added to the device configuration file and the propagated to the Nuimo app config.

2) On startup, Nuimo app would either create a group of lights (if # of lights > 10) and control that or otherwise control each light separately.

Also adding the Philips Hue bridge username to the device configuration file for convenience.

larsblumberg commented:
As we just discussed, we want to send the ON/OFF command always to a group to have all lights (if more than one) switch on/off at the very same time

larsblumberg commented on senic_hub/nuimo_app/components/philips_hue.py:
```diff
@@ -1,188 +1,351 @@
 
-    # how often to sync with the bridge to detect changes made by external apps
-    SYNC_INTERVAL = 5  # seconds
+        merged = self.merge_success_responses(responses)
```
`merged` reads like a boolean variabe to name. Since we don't annotate with types, we should use expressive names that suggest their type or at least don't suggest misleading types. Here maybe: `merged_response`?

grunskis replied:
Addressed: 47994b8

larsblumberg commented on senic_hub/nuimo_app/components/philips_hue.py:
```diff
@@ -1,188 +1,351 @@
+        return 60  # seconds
+
+
+class Lights(HueBase):
```
I wouldn't use plural name for classes, here I would use `LightCollection` or `LightSet`.

grunskis replied:
Addressed: 4b201f6

larsblumberg commented on senic_hub/nuimo_app/components/philips_hue.py:
```diff
@@ -1,188 +1,351 @@
+
+        logger.debug("state: %s", pformat(self._state))
+
+        self._on = any(s['on'] for s in self._state.values())
```
Philips Hue bridge considers a set of lights to be off if at least one if their lights is off. This would be contrary to the formula here where you assume all lights on if at least one is on. This said, it should probably be:

    self._on = all(s['on'] for s in self._state.values())

grunskis replied:
Thanks! Addressed ba8649b

larsblumberg commented on senic_hub/nuimo_app/components/philips_hue.py:
```diff
@@ -1,188 +1,351 @@
-        if group_lights != all_lights:
-            return self.update_light_group(group_id, lights)
+    def create_group(self):
+        responses = self.bridge.create_group(self.GROUP_NAME, self.light_ids)
```
Is `create_group` returning multiple responses or a single response? The variable name `responses` suggests that we have multiple responses.

grunskis replied:
Yep, it returns list of responses even though there's usually one item in the list. It's super annoying...

grunskis commented on senic_hub/backend/views/setup_devices.py:
```diff
@@ -152,11 +152,18 @@ def get_device(device_list_path, device_id):
     with open(devices_path, "r") as f:
         devices = json.loads(f.read())
 
+    device["extra"]["username"] = username
```
Ok. I'm still getting used to single quotes :)

grunskis commented:
I addressed your feedback. Please check...

I also tried to improve the value normalization/clamping 1afb428 Please take a look and give feedback...

The ON/OFF logic with multiple lights I'll address in a separate PR.

grunskis commented:
Thanks for suggestion. I implemented a version of this here c9fa9eb IMO is much cleaner now...
